### PR TITLE
Build rhel package for s390x

### PR DESCRIPTION
**Pull request by StefanScherer**
Signed-off-by: Stefan Scherer <stefan.scherer@docker.com>

### CODE REVIEW

File: dockerfiles/rpm.dockerfile [hunk 1]
```diff
@@ -72,6 +72,7 @@ ENV GOPATH=/go
 ENV PATH="${PATH}:/usr/local/go/bin:${GOPATH}/bin"
 ENV IMPORT_PATH=github.com/containerd/containerd
 ENV GO_SRC_PATH="/go/src/${IMPORT_PATH}"
+ENV CC=gcc
```
Comment (StefanScherer) [inline dockerfiles/rpm.dockerfile:75]: This fixes an issue build the package on s390x with the clefos:7 image

```
exec: "s390x-linux-gnu-gcc": executable file not found in $PATH
```

Comment (justincormack) [reply dockerfiles/rpm.dockerfile:75]: this should be fine, Go only builds with GCC anyway.

Comment (StefanScherer) [reply dockerfiles/rpm.dockerfile:75]: Cool, thanks. I'll squash the commits.

**Comment by StefanScherer** [comment]
Build for `clefos:7-s390x` is green -> https://ci-next.docker.com/teams-core/blue/organizations/jenkins/containerd-packaging/detail/PR-167/4/pipeline/78, other errors are unrelated to this PR.

File: Jenkinsfile [hunk 2]
```diff
@@ -9,6 +9,7 @@ def images = [
     [image: "amazonlinux:2",                  arches: ["aarch64"]],
     [image: "centos:7",                       arches: ["amd64", "aarch64", "armhf"]],
     [image: "centos:8",                       arches: ["amd64", "aarch64"]],          // Note: armhf (arm32) images are currently not available on Docker Hub
+    [image: "clefos:7",                       arches: ["s390x"]],
```
Comment (thaJeztah) [inline Jenkinsfile:12]: Should these packages end up in the `centos` directories on download.docker.com, or in `clefos` ? If they need to go into `centos`, so we need some handling to rewrite the paths?

Comment (StefanScherer) [reply Jenkinsfile]: Ha, you tricked me 😅 I tried to modify the Jenkinsfile to tag clefos:7 -> centos:7 and sorts of things, but ran into more problems now. I looked at [that specific build](https://ci-next.docker.com/teams-core/blue/organizations/jenkins/containerd-packaging/detail/PR-167/4/pipeline/78#step-470-log-975) with just use `image: "clefos:7"` and I found that the output is already in a `centos/7` sub folder, the BASE variable is automatically set to `centos`.
I'm changing back to the initial change.

But at least I now have rebased the PR :-)

**Comment by StefanScherer** [comment]
After the rebase I now have this problem

```
#30 2.725 Error: Package: containerd.io-0.20200420.163226~1d083fec-0.el7.s390x (local)
#30 2.725            Requires: container-selinux >= 2:2.74
#30 2.725            Available: 2:container-selinux-1.12.3-5.git9a594b9.el7.s390x (epel)
#30 2.725                container-selinux = 2:1.12.3-5.git9a594b9.el7
#30 2.725            Available: 2:container-selinux-2.2-3.el7.noarch (epel)
#30 2.725                container-selinux = 2:2.2-3.el7
```

**Comment by StefanScherer** [comment]
I'm not a CentOS user, so what I found out so far is
in the clefos:7 image running `yum` commands I often get this error

```
https://download.sinenomine.net/clefos/7/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
```

In https://www.sinenomine.net/products/linux/clefos/packages/clefos7-bin I can see newer versions in the `noarch` section:

clefos-container-selinux

```
bash-4.2# yum repolist 
Loaded plugins: fastestmirror, ovl
Loading mirror speeds from cached hostfile
https://download.sinenomine.net/clefos/7/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
Trying other mirror.
To address this issue please refer to the below wiki article 

https://wiki.centos.org/yum-errors

If above article doesn't help to resolve this issue please use https://bugs.centos.org/.

https://download.sinenomine.net/clefos/7/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
Trying other mirror.
repo id                                            repo name                                                                  status
epel                                               Extra Packages for Enterprise Linux 7 -                                    46839
os/7/s390x                                         ClefOS-7 - Base                                                                0
oss                                                Sine Nomine Associates packages                                             1760
updates/7/s390x                                    ClefOS-7 - Updates                                                          1902
repolist: 50501
bash-4.2# yum --disablerepo=os list container-selinux
Loaded plugins: fastestmirror, ovl
Loading mirror speeds from cached hostfile
Available Packages
container-selinux.s390x                                        2:1.12.3-5.git9a594b9.el7                                        epel
container-selinux.noarch                                       2:2.2-3.el7                                                      epel
```

**Comment by thaJeztah** [comment]
I assume `noarch` should be fine for the selinux profiles/policies (what's the right name?), at least I think those should not be architecture dependent?

**Comment by StefanScherer** [comment]
Updated the PR to use the RHEL ubi base image, the clefos had too many 404 errors.

File: dockerfiles/rpm.dockerfile (hunk 1 above)
Comment (StefanScherer) [reply dockerfiles/rpm.dockerfile:67]: This is still needed for the ubi base image.

**Comment by StefanScherer** [comment]
Rebased and added rhel 8 as well. All is green now :-)

File: Jenkinsfile [hunk 3]
```diff
@@ -8,20 +8,21 @@
...
-    [image: "docker.io/library/ubuntu:xenial",          arches: ["amd64", "aarch64", "armhf", "s390x"]], // Ubuntu 16.04 LTS (End of support: April, 2021. EOL: April, 2024)
+    [image: "docker.io/library/ubuntu:xenial",          arches: ["amd64", "aarch64", "armhf"]], // Ubuntu 16.04 LTS (End of support: April, 2021. EOL: April, 2024)
     [image: "docker.io/library/ubuntu:bionic",          arches: ["amd64", "aarch64", "armhf", "s390x"]], // Ubuntu 18.04 LTS (End of support: April, 2023. EOL: April, 2028)
-    [image: "docker.io/library/ubuntu:eoan",            arches: ["amd64", "aarch64", "armhf", "s390x"]], // Ubuntu 19.10  (EOL: July, 2020)
-    [image: "docker.io/library/ubuntu:focal",           arches: ["amd64", "aarch64", "s390x"]],          // Ubuntu 20.04 LTS (End of support: April, 2025. EOL: April, 2030)
+    [image: "docker.io/library/ubuntu:eoan",            arches: ["amd64", "aarch64", "armhf"]], // Ubuntu 19.10  (EOL: July, 2020)
+    [image: "docker.io/library/ubuntu:focal",           arches: ["amd64", "aarch64"]],          // Ubuntu 20.04 LTS (End of support: April, 2025. EOL: April, 2030)
```
Comment (thaJeztah) [inline Jenkinsfile:25]: Guess we should add `ubuntu:focal` as well (being the next LTS release); ok for a follow-up if needed though

Comment (StefanScherer) [reply Jenkinsfile:25]: Yes, could be done in a follow-up PR once we need that.

**Pull request merged**